### PR TITLE
Exit with an error when building docs if a notebook fails to execute

### PR DESCRIPTION
**Pull request by otizonaizit**
After merging this PR the CI will fail if a notebook in the docs fails to execute. This way we can be sure that the docs are always up-to-date with the latest changes in the code.

To achieve this I had to adjust a bit the doc-building procedure.

Important: 
- I disabled the building of https://github.com/wichmann-lab/python-psignifit/blob/main/docs/examples/priors.md because it is still using the `stimulus_range` option that was removed in #171 .
- I removed the obsolete demo notebooks as discussed in #180 

Fixes: #180

### CODE REVIEW

**Comment by guillermoaguilar** [comment]
All good except that this PR is deleting the demos

- parameter_recovery_in_logspace_demo.ipynb
- parameters_estimates.ipynb

But these have not been migrated yet.

What you can do is just move them to docs/to-be-migrated   or something
and then add this folder to be excluded from docs build

File: psignifit/demos/parameter_recovery_in_logspace_demo.ipynb
Comment (guillermoaguilar) [inline psignifit/demos/parameter_recovery_in_logspace_demo.ipynb:1]: keep this one, move to docs/not-yet-migrated

File: psignifit/demos/parameters_estimates.ipynb
Comment (guillermoaguilar) [inline psignifit/demos/parameters_estimates.ipynb:1]: this one too

**Comment by otizonaizit** [comment]
> All good except that this PR is deleting the demos
> 
>     * parameter_recovery_in_logspace_demo.ipynb
> 
>     * parameters_estimates.ipynb
> 
> 
> But these have not been migrated yet.

OK, I reinstated the two non ported demos

**Comment by guillermoaguilar** [comment]
LGTM!

**Pull request merged**